radixdlt/src/main/java/org/radix/api/AtomQuery.java

```diff
@@ -17,18 +17,13 @@
 
 package org.radix.api;
 
-import com.radixdlt.atomos.RadixAddress;
 import com.radixdlt.common.EUID;
 
 import java.util.Set;
 
-public class AtomQuery {
+public final class AtomQuery {
 	private final EUID destination;
 
-	public AtomQuery(RadixAddress address) {
```

Comment:
This constructor makes no sense to me

Reply:
Why? It probably shouldn't be a constructor (an AtomQuery.from method seems more adequate), but the functionality makes sense to me.. what am I missing?

Reply:
The implementation of this constructor is essentially `this(address.getUID())` although it isn't implemented that way.

This makes the dependency here on RadixAddress completely unnecessary.